plugins/search/src/components/SearchBar/SearchBar.tsx

```diff
@@ -30,16 +30,25 @@ type PresenterProps = {
   onSubmit?: () => void;
   className?: string;
   placeholder?: string;
+  focused?: boolean;
 };
 
 export const SearchBarBase = ({
+  focused,
```

Comment:
Was thinking: what about if we [matched MUI](https://mui.com/api/input/#props) `autoFocus`, but that got me thinking...  Can we just pass that down to the `<InputBase />` instead of dealing in refs and `useEffect`s?

Reply:
yeah I was a bit confused by that... as it says when using it:

`The autoFocus prop should not be used, as it can reduce usability and accessibility for users.`

Reply:
@iamEAP I guess the alternative is - make it possible for people. So when you compose your app its up to you to decide? But not have it as a default on any components?

Reply:
*(Surfaces from auto-focus / accessibility rabbit hole)*.

Yeah, it seems like there are some valid, accessible use-cases for auto-focus (e.g. the main search field on the Google homepage), but many others that are bad for accessibility / UX.  So giving the choice (and definitely not defaulting on) is good.

It also seems like the implementation you have here (`useRef` + `.focus()`) is [functionally equivalent](https://stackoverflow.com/a/60216924/12064302) to the react `autoFocus` prop (note: uppercase `F`), which is distinct from the HTML `autofocus` attribute ([e.g.](https://blog.maisie.ink/react-ref-autofocus)).

So I'm thinking we could kill two birds with one stone here by using `autoFocus` as the prop name and passing its value down to `<InputBase />`:

- No need to re-implement the logic ourselves w/`useEffect()` + `useRef()`.
- It's a tiny bit more idiomatic for react users.

Whaddya think @emmaindal?

Reply:
Yes, totally agree! So re-using autoFocus but make it a optional prop. Default to not using it! Thanks for input @iamEAP